Partition (WBP)/Functions/VI0049C.VTI0049.fun.yml

```diff
@@ -0,0 +1,305 @@
+﻿Function:
+  Enabled For RDMLX:  Yes
+
+  Source: |
+    Function Options(*DIRECT) Rcv_List(#wLKData #wLRet)
+    * ******************************************************************************************************
+    * Test Case Description
+    * Issue #49 Performance Test - Using Literal and Bind Variables
+    * The test setup includes a new table named VTLI0049 with Char and Integer columns
+    * This test case will be utilized by LANSA developer to obtain base line performance statistics
+    * and then database-specific SQL to act over that table using bind variables
+    * ******************************************************************************************************
+    * Unique Key Data.
+    Def_List Name(#wLKData) Fields(#VTF_KEY1 #VTF_KEY2 #VTF_KEY3) Type(*WORKING) Entrys(1)
+    
+    * Return messages.
+    Def_List Name(#wLRet) Fields(#BIFMSGC #BIFMSG1 #BIFMSG2) Counter(#BIFMSGCNT) Type(*WORKING) Entrys(*MAX)
+    
+    Define Field(#LF_i) Reffld(#Std_Count)
+    Define Field(#LF_Count) Reffld(#Std_Count)
+    Define Field(#LF_rc) Reffld(#BIFA002A) Default(OK)
+    Define Field(#LF_logfile_name) Reffld(#BIFA256A)
+    
+    Define Field(#LF_DB_Type) Type(*CHAR) Length(32)
+    Use Builtin(Get_Session_Value) With_Args(DBUT) To_Get(#LF_DB_Type #LF_rc)
+    
+    * Start testing.
+    Clr_List Named(#wLRet)
+    #Com_Owner.mthAdd_Msg( OK *Function ('Start - ' + *Function_Desc) )
+    #Com_Owner.mthGetExecutionType()
+    #Com_Owner.mthAdd_SubMsg( OK 'Database Type' (#LF_DB_Type) )
+    
+    *
+    * Tests are only to be executed when run from within the MSSQL System.
+    * Only on Windows 32 bit.
+    * The MSSQL Test is only executed when its all local, not in SuperServer mode.
+    * All the other databases are ONLY run when running as a SuperServer Function (not local and not SuperServer IO).
+    *
+    If (*OSBITNESS *EQ 32)
+    * If ((#LF_Execution_Type *EQ W) *And (#LF_Local_Platform *EQ W) *And (#LF_DB_Type *EQ MSSQLS) OrIf (#LF_Execution_Type *EQ F) *And (#LF_Local_Platform *EQ W) *And (#LF_Remote_Platform *EQ W) *And (#LF_DB_Type *NE MSSQLS))
+    If (((#LF_Execution_Type *EQ W) *And (#LF_Local_Platform *EQ W) *And (#LF_DB_Type *EQ MSSQLS)) OrIf ((#LF_Execution_Type *EQ F) *And (#LF_Local_Platform *EQ W) *And (#LF_Remote_Platform *EQ W) *And (#LF_DB_Type *NE MSSQLS)))
+    #com_owner.mthDoDelete( #LF_rc )
+    #Com_Owner.mthInsert( #LF_rc )
+    
```

Comment:
Now that we have the whole set of data controlled by the Local Windows Function, we can delete the log file when that Function runs:
```
if ((#LF_Execution_Type *EQ W) *And (#LF_Local_Platform *EQ W) *And (#LF_DB_Type *EQ MSSQLS))
<delete #LF_logfile_name>
endif
```
Note: Set the value of the log file once

Reply:
If I delete #LF_logfile_name from VI0049C after  #Com_Owner.mthExecuteSqlScript(#Lf_rc )
, Can you please let me know how can I utilize the log file for TRANSFORM_LIST in VI0049D within the routine mthTransformPerformanceMetricsLogIntoTable?

I understand , I have to read the contents of the log file into a list named for ex: #w_lTL using TRANSFORM_LIST and then pass this list to TRANSFORM_File
Is my understanding correct ?

Reply:
Delete the file BEFORE #Com_Owner.mthExecuteSqlScript(#Lf_rc ).

Other way around. TRANSFORM_FILE transforms a file into a list. Then TRANSFORM_LIST transforms a list into a file.

Reply:
I will be fixing this in the next commit. Thank you

Reply:
I fixed the delete of the log file .  Implementation of TRANSFORM_LIST and TRANSOFRM_FILE is still in progress.

Reply:
@robe070 , the delete of the logfile is overwriting for every database run. For ex: I run for MSSQLS,log file gets generated with MSSQL performance metrics, immediately we run for MySQL, MSSQLS performance metrics is overwritten with MYSQL content and so on.... Can we remove the deletion of the logfile?

Reply:
Fixed.